tests/unit/block_device_test.cc

```diff
@@ -68,6 +72,95 @@ static future<> test_basic_read_write(const std::string& device_path) {
     });
 }
 
+static future<> test_parallel_read_write(const std::string& device_path) {
+    return async([&] {
+        block_device dev = open_block_device(device_path).get0();
+        constexpr size_t buff_size = 16 << 20;
+        auto buffer = allocate_random_aligned_buffer(buff_size);
+
+        // Write
+        static_assert(buff_size % alignment == 0);
+        parallel_for_each(boost::irange<off_t>(0, buff_size / alignment), [&](off_t block_no) {
+            off_t offset = block_no * alignment;
+            return dev.write(offset, buffer.get() + offset, alignment).then([](size_t written) {
+                assert(written == alignment);
+            });
+        }).get0();
+
+        // Read
+        static_assert(buff_size % alignment == 0);
+        parallel_for_each(boost::irange<off_t>(0, buff_size / alignment), [&](off_t block_no) {
+            return async([&dev, &buffer, block_no] {
+                off_t offset = block_no * alignment;
+                auto check_buffer = allocate_random_aligned_buffer(alignment);
+                assert(dev.read(offset, check_buffer.get(), alignment).get0() == alignment);
+                assert(memcmp(buffer.get() + offset, check_buffer.get(), alignment) == 0);
+            });
+        }).get0();
+
+        dev.close().get0();
+    });
+}
+
+static future<> test_simultaneous_parallel_read_and_write(const std::string& device_path) {
+    return async([&] {
+        block_device dev = open_block_device(device_path).get0();
+        constexpr size_t buff_size = 16 << 20;
+        auto buffer = allocate_random_aligned_buffer(buff_size);
+        assert(dev.write(0, buffer.get(), buff_size).get0() == buff_size);
+
+        static_assert(buff_size % alignment == 0);
+        size_t blocks = buff_size / alignment;
```

Comment:
for a variable that represents a size, use `blocks_num` or `blocks_count` - `blocks` suggests that it's a data structure with blocks inside